services/src/main/scala/org/apache/gearpump/services/MasterService.scala

```diff
@@ -98,14 +97,16 @@ trait MasterService extends HttpService {
       } ~
       path("submitapp") {
         post {
-          respondWithMediaType(MediaTypes.`application/json`) {
-            entity(as[MultipartFormData]) { formData =>
-              val jar = formData.fields.head.entity.data.toByteArray
-              onComplete(Future(MasterService.submitJar(jar, system.settings.config))) {
-                case Success(_) =>
-                  complete(write(MasterService.Status(true)))
-                case Failure(ex) =>
-                  failWith(ex)
+          anyParams('args.as[Option[String]]) { (extraArgs) =>
```

Comment:
There will not have this concern, in case we split it as two APIs: one for upload a jar; one for launch the jar with specified arguments.